libdnf/module/ModulePackage.cpp

```diff
@@ -0,0 +1,182 @@
+                    if (moduleStream.find('-', 0) != std::string::npos) {
+                        ss << ":" << moduleStream.substr(1) << ")";
+                        depId = pool_str2id(pool, ss.str().c_str(), 1);
+                        solvable_add_deparray(solvable, SOLVABLE_CONFLICTS, depId, -1);
+                    } else {
+                        hasStreamRequires = true;
+                        ss << ":" << moduleStream << ")";
+                        depId = pool_str2id(pool, ss.str().c_str(), 1);
+                        solvable_add_deparray(solvable, SOLVABLE_REQUIRES, depId, 0);
+                    }
+                }
+
+                if (!hasStreamRequires) {
+                    // without stream; just close parenthesis
+                    ss << ")";
+                    depId = pool_str2id(pool, ss.str().c_str(), 1);
+                    solvable_add_deparray(solvable, SOLVABLE_REQUIRES, depId, -1);
+                }
+            }
+        }
+    }
+}
+
+std::string ModulePackage::getName() const
+{
+    return metadata->getName();
+}
+
+std::string ModulePackage::getNameStream() const
+{
+    std::ostringstream ss;
+    ss << getName() << ":" << getStream();
+    return ss.str();
+}
+
+std::string ModulePackage::getNameStreamVersion() const
+{
+    std::ostringstream ss;
+    ss << getNameStream() << ":" << getVersion();
+    return ss.str();
+}
+
+std::string ModulePackage::getStream() const
+{
+    return metadata->getStream();
+}
+
+std::string ModulePackage::getVersion() const
+{
+    return std::to_string(metadata->getVersion());
+}
+
+std::string ModulePackage::getContext() const
+{
+    return metadata->getContext();
+}
+
+std::string ModulePackage::getArchitecture() const
+{
+    return metadata->getArchitecture();
+}
+
+std::string ModulePackage::getFullIdentifier() const
+{
+    std::ostringstream ss;
+    ss << getName() << ":" << getStream() << ":" << getVersion() << ":" << getContext() << ":" << getArchitecture();
+    return ss.str();
+}
+
+std::string ModulePackage::getSummary() const
+{
+    return metadata->getSummary();
+}
+
+std::string ModulePackage::getDescription() const
+{
+    return metadata->getDescription();
+}
+
+std::vector<std::string> ModulePackage::getArtifacts() const
+{
+    return metadata->getArtifacts();
+}
+
+std::shared_ptr<Profile> ModulePackage::getProfile(const std::string &name) const
+{
+    return metadata->getProfile(name);
+}
+
+std::vector<std::shared_ptr<ModuleProfile> > ModulePackage::getProfiles() const
+{
+    return metadata->getProfiles();
+}
+
+std::vector<std::shared_ptr<ModuleDependencies> > ModulePackage::getModuleDependencies() const
+{
+    return metadata->getDependencies();
+}
+
+bool ModulePackage::isEnabled()
+{
+    return state->isEnabled();
+}
+
+void ModulePackage::setState(const std::shared_ptr<ModuleState> &state)
+{
+    this->state = state;
+}
+
+void ModulePackage::enable()
```

Comment:
ModulePackage represents a modulemd module record. A stream consists of these records. Running enable() on a single record doesn't make sense to me.
Enable() should be a standalone function or a method of some kind of container.